### PR TITLE
one big chunk of code in a feature branch

### PR DESCRIPTION
**Pull request by Elhambasir**
This pull request is made for the coding partner's project review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,268 @@
...
+- Button Interactions (i.e. hover, etc.)
+- Attractive Images & Design
+- Modal/dialog
+- Dynamic page (data is retrieved from JSON file)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 🚀 Live Demo <a name="live-demo"></a>
+[Live Demo Link](https://Elhambasir.github.io/To-Do-List/)
```
Comment (Lul-Abdifan) [inline README.md]: - Nice to have a Live Demo, Could you please fix this line of code? It is directing me to your web. or Add server.js file in order review your project.
image

**Review by Lul-Abdifan — CHANGES_REQUESTED** [review]
Hello @Elhambasir, Good job so far!

Highlights

-  Professional Readme File :100: 
-  Linters are setup correctly :100: 
-  Clear and nice code :100: 



There is some hindrance that makes me not review the UI of your project.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.

File: README.md (hunk 1 above)
Comment (Elhambasir) [reply README.md]: Thank you I will fix that as soon as possible

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+<body>
+  <main class="container">
+    <div class="sub-container">
+      <div class="title">
+        <h3>Today's To Do</h3>
+        <button><i class="fas fa-refresh"></i></button>
+      </div>
+      <hr class="hr">
+      <form action="" id="job-form">
+        <input type="text" name="job" class="input-job" placeholder="Add to your list">
```
Comment (Lul-Abdifan) [inline src/index.html:22]: Kindly add a label to the form input elements to ensure you comply with HTML best practices, i.e accessibility. :smile:

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+const taskObject1 = new Task();
+const elementObject1 = new Element();
+elementObject1.btnAddNew.addEventListener('click', () => {
+  if (elementObject1.titleInput.value) {
+    taskObject1.addTask();
+  }
+});
+taskObject1.retrieveData();
+window.addEventListener('load', () => {
+  document.querySelectorAll('.checkbox').forEach((element) => {
```
Comment (Lul-Abdifan) [inline src/index.js:16]: -  Could you please initiate all variables at the beginning of the various scopes in order to keep JavaScript best practices

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    if (element.attributes.value === 'checked') {
+      element.classList.add('overlined');
+    } else {
+      element.classList.remove('overlined');
+    }
+  });
+});
+
+document.querySelectorAll('.checkbox').forEach((element) => {
```
Comment (Lul-Abdifan) [inline src/index.js:25]: - Could you please also look at this?

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  element.nextElementSibling.addEventListener('click', (e) => {
+    e.target.parentElement.setAttribute('style', 'background-color:lightgray;');
+    e.target.nextElementSibling.classList.remove('fa-trash', 'fa-xs', 'fa-grip-lines');
+    e.target.nextElementSibling.classList.add('fa-trash', 'fa-xs', 'delete');
+    e.target.nextElementSibling.addEventListener('click', () => {
+      taskObject1.removeItem(element.getAttribute('index'));
+    });
+
+    document.querySelectorAll('.checkbox').forEach((element) => {
```
Comment (Lul-Abdifan) [inline src/index.js:34]: - Could you please also look at this?

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      if (element.nextElementSibling !== e.currentTarget) {
+        element.parentElement.setAttribute('style', 'background-color:white;');
+        element.nextElementSibling.nextElementSibling.classList.add('fa-trash', 'fa-xs', 'fa-grip-lines');
+        element.nextElementSibling.nextElementSibling.classList.remove('fa-trash', 'fa-xs', 'delete');
+      }
+    });
+  });
+});
+document.querySelectorAll('.checkbox').forEach((element) => {
```
Comment (Lul-Abdifan) [inline src/index.js:43]: - Could you please also look at this?

**Review by Lul-Abdifan — APPROVED** [review]
Hello @Elhambasir, I appreciate your Job. Your project is complete! There is nothing else to say other than... it's time to merge it :clap: :clap: :clap: 
Congratulations! 



## Highlights
✅ Good Implementation 
✅ Professional PR
✅ Professional Commits
✅ No Linters error

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**